deployment/common/changeset/set_config_mcms.go

```diff
@@ -0,0 +1,207 @@
+	bypasserTx  *types.Transaction
+}
+
+// setConfigPerRole sets the configuration for each of the MCMS contract roles on the mcmsState.
+func setConfigPerRole(chain deployment.Chain, cfg ConfigPerRole, mcmsState *MCMSWithTimelockState, useMCMS bool) (setConfigTxs, error) {
+	// Proposer set config
+	proposerTx, err := setConfigOrTxData(chain, cfg.Proposer, mcmsState.ProposerMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+	// Canceller set config
+	cancellerTx, err := setConfigOrTxData(chain, cfg.Canceller, mcmsState.CancellerMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+	// Bypasser set config
+	bypasserTx, err := setConfigOrTxData(chain, cfg.Bypasser, mcmsState.BypasserMcm, useMCMS)
+	if err != nil {
+		return setConfigTxs{}, err
+	}
+
+	return setConfigTxs{
+		proposerTx:  proposerTx,
+		cancellerTx: cancellerTx,
+		bypasserTx:  bypasserTx,
+	}, nil
+}
+
+func addTxsToProposalBatch(setConfigTxsChain setConfigTxs, chainSelector uint64, state MCMSWithTimelockState) timelock.BatchChainOperation {
+	result := timelock.BatchChainOperation{
+		ChainIdentifier: mcms.ChainIdentifier(chainSelector),
+		Batch:           []mcms.Operation{},
+	}
+	result.Batch = append(result.Batch, mcms.Operation{
+		To:           state.ProposerMcm.Address(),
+		Data:         setConfigTxsChain.proposerTx.Data(),
+		Value:        big.NewInt(0),
+		ContractType: string(commontypes.ProposerManyChainMultisig),
+	})
+	result.Batch = append(result.Batch, mcms.Operation{
+		To:           state.CancellerMcm.Address(),
+		Data:         setConfigTxsChain.cancellerTx.Data(),
+		Value:        big.NewInt(0),
+		ContractType: string(commontypes.CancellerManyChainMultisig),
+	})
+	result.Batch = append(result.Batch, mcms.Operation{
+		To:           state.BypasserMcm.Address(),
+		Data:         setConfigTxsChain.bypasserTx.Data(),
+		Value:        big.NewInt(0),
+		ContractType: string(commontypes.BypasserManyChainMultisig),
+	})
+	return result
+}
+
+// SetConfigMCMS sets the configuration of the MCMS contract on the chain identified by the chainSelector.
+func SetConfigMCMS(e deployment.Environment, cfg SetConfigParams) (deployment.ChangesetOutput, error) {
+	selectors := []uint64{}
+	for chainSelector := range cfg.ConfigsPerChain {
+		selectors = append(selectors, chainSelector)
+	}
+	useMCMS := cfg.ProposalConfig != nil
+	err := cfg.Validate(e, selectors)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+
+	batches := []timelock.BatchChainOperation{}
+	timelocksPerChain := map[uint64]common.Address{}
+	proposerMcmsPerChain := map[uint64]*gethwrappers.ManyChainMultiSig{}
+
+	mcmsStatePerChain, err := MaybeLoadMCMSWithTimelockState(e, selectors)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+
+	for chainSelector, c := range cfg.ConfigsPerChain {
+		chain, ok := e.Chains[chainSelector]
+		if !ok {
+			return deployment.ChangesetOutput{}, errors.New("chain not found in environment")
```

Comment:
this is already asserted in the Validate

Reply:
removed here https://github.com/smartcontractkit/chainlink/pull/15684/commits/0f4cbf369a6f66423b89a2ffdd68add367691c1e